src/app/chat/views/channels_list.nim

```diff
@@ -117,7 +117,11 @@ QtObject:
 
     result = self.chats.len
 
-  proc getChannel*(self: ChannelsList, index: int): Chat = self.chats[index]
+  proc getChannel*(self: ChannelsList, index: int): Chat = 
+    if index < 0 or index >= self.chats.len:
+      return
```

Comment:
I'm curious here of what happens when you return early? Since this proc returns a Chat, wouldnt it return then empty Chat object? Could this be a problem?

cc: @jrainville

Reply:
@richard-ramos yes, that's is exactly the point. You must ensure that accessing an element by index is done within the lower limit (0) and upper limit (sequence length - 1), otherwise you will get a crash.

For this getChannel proc, nil object will be returned and it's on the caller to check if the returned value is nil.

Reply:
I don't think it will be `nil`. Nim usually returns an empty object in that case, so the check you'd need to do on the caller side is something like `chat.id == ""`.

Reply:
@jrainville we discussed this in private chat, but just to conclude here. Chat is a ref object and that's why `nil` will be returned.